operators/test/e2e/cmd/run/command.go

```diff
@@ -0,0 +1,89 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package run
+
+import (
+	"math/rand"
+	"strings"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+type runFlags struct {
+	managedNamespaces   []string
+	e2eImage            string
+	elasticStackVersion string
+	kubeConfig          string
+	operatorImage       string
+	testContextOutPath  string
+	testLicence         string
+	testOutDirRoot      string
+	testRegex           string
+	testRunName         string
+	commandTimeout      time.Duration
+	autoPortForwarding  bool
+	setupOnly           bool
+	skipCleanup         bool
+}
+
+var log logr.Logger
+
+func Command() *cobra.Command {
+	flags := runFlags{}
+
+	cmd := &cobra.Command{
+		Use:   "run",
+		Short: "setup an e2e test environment and run tests",
+		PreRun: func(_ *cobra.Command, _ []string) {
+			log = logf.Log.WithName(flags.testRunName)
+		},
+		RunE: func(_ *cobra.Command, _ []string) error {
+			if flags.setupOnly {
+				log.Info("Disabling cleanup because this is a setup-only run")
+				flags.skipCleanup = true
+			}
+			return doRun(flags)
+		},
+	}
+
+	cmd.Flags().BoolVar(&flags.autoPortForwarding, "auto-port-forwarding", false, "Enable port forwarding to pods")
+	cmd.Flags().DurationVar(&flags.commandTimeout, "command-timeout", 90*time.Second, "Timeout for commands executed")
+	cmd.Flags().StringVar(&flags.e2eImage, "e2e-image", "", "E2E test image")
+	cmd.Flags().StringVar(&flags.elasticStackVersion, "elastic-stack-version", "7.1.1", "Elastic stack version")
+	cmd.Flags().StringVar(&flags.kubeConfig, "kubeconfig", "", "Path to kubeconfig")
+	cmd.Flags().StringSliceVar(&flags.managedNamespaces, "managed-namespaces", []string{"mercury", "venus"}, "List of managed namespaces")
```

Comment:
They get prefixed with the test run name (Eg. `e2e-1kdkq-mercury`). I thought about using `alpha`, `beta` etc. but that could confuse people because they might assume that those names refer to versions. `a`, `b` or `1`, `2` are too short so I decided to use the order of planets in the solar system instead. :smile:

Reply:
hah, got it 😄